compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -8245,7 +8245,14 @@ private STNode parseFieldIdent() {
     private STNode parseXMLNamespaceDeclaration(boolean isModuleVar) {
         startContext(ParserRuleContext.XML_NAMESPACE_DECLARATION);
         STNode xmlnsKeyword = parseXMLNSKeyword();
-        STNode namespaceUri = parseXMLNamespaceUri();
+
+        STNode namespaceUri = parseSimpleConstExpr();
+        while (!isValidXMLNameSpaceURI(namespaceUri)) {
+            xmlnsKeyword = SyntaxErrors.cloneWithTrailingInvalidNodeMinutiae(xmlnsKeyword,
+                    namespaceUri, DiagnosticErrorCode.ERROR_INVALID_XML_NAMESPACE_URI);
+            namespaceUri = parseSimpleConstExpr();
```

Comment:
Trying to recover again can be expensive..
Since we have already parsed the expression and knows that it is invalid, we can create a dummy literal/var-ref here and use that as the uri.

Reply:
Then `while` can be a `if`

Reply:
but what if there is a valid uri next to it eg: `xmlns 5 "myxml" as ns1;`

Reply:
I see your point.
+1 for the current approach then.